core/controllers/admin.py

```diff
@@ -635,6 +636,168 @@ def get(self):
+        new_reviewer_user_id = (
+            user_services.get_user_id_from_username(new_reviewer_username))
+
+        if new_reviewer_user_id is None:
+            raise self.InvalidInputException(
+                'Invalid username: %s' % new_reviewer_username)
+
+        review_category = self.payload.get('review_category')
+        language_code = self.payload.get('language_code', None)
+
+        if review_category == constants.REVIEW_CATEGORY_TRANSLATION:
+            if not utils.is_supported_audio_language_code(language_code):
+                raise self.InvalidInputException(
+                    'Invalid language_code: %s' % language_code)
+            if user_services.can_review_translation_suggestions(
+                    new_reviewer_user_id, language_code=language_code):
+                raise self.InvalidInputException(
+                    'User %s already has rights to review translation in '
+                    'language code %s' % (
+                        new_reviewer_username, language_code))
+            user_services.allow_user_to_review_translation_in_language(
+                new_reviewer_user_id, language_code)
+        elif review_category == constants.REVIEW_CATEGORY_VOICEOVER:
+            if not utils.is_supported_audio_language_code(language_code):
+                raise self.InvalidInputException(
+                    'Invalid language_code: %s' % language_code)
+            if user_services.can_review_voiceover_applications(
+                    new_reviewer_user_id, language_code=language_code):
+                raise self.InvalidInputException(
+                    'User %s already has rights to review voiceover in '
+                    'language code %s' % (
+                        new_reviewer_username, language_code))
+            user_services.allow_user_to_review_voiceover_in_language(
+                new_reviewer_user_id, language_code)
+        elif review_category == constants.REVIEW_CATEGORY_QUESTION:
+            if user_services.can_review_question_suggestions(
+                    new_reviewer_user_id):
+                raise self.InvalidInputException(
+                    'User %s already has rights to review question.' % (
+                        new_reviewer_username))
+            user_services.allow_user_to_review_question(new_reviewer_user_id)
+        else:
+            raise self.InvalidInputException(
+                'Invalid review_category: %s' % review_category)
+
+        email_manager.send_email_to_new_community_reviewer(
+            new_reviewer_user_id, review_category, language_code=language_code)
+        self.render_json({})
+
+
+class RemoveCommunityReviewerHandler(base.BaseHandler):
+    """Handles removing reviewer for community dashboard."""
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        username = self.payload.get('username')
```

Comment:
Here and elsewhere, I think you need checks to ensure that the keys exist, or default them to None and treat those cases explicitly by raising a 400.

Reply:
For no value the default value is None and I have handled the case for None below
![image](https://user-images.githubusercontent.com/16653571/76311139-4bb74400-62f6-11ea-9dd4-f424939a4d86.png)